src/meta/src/hummock/manager/mod.rs

```diff
@@ -1109,136 +1179,222 @@ where
+        );
+
+        new_hummock_version = old_version;
+        new_version_delta1.id = new_version_id;
+        new_hummock_version.id = new_version_id;
+        Ok((
+            new_version_delta1.key,
+            new_version_delta1.new_value,
+            new_hummock_version,
+        ))
     }
 
     /// Caller should ensure `epoch` > `max_committed_epoch`
     #[named]
     pub async fn commit_epoch(
         &self,
         epoch: HummockEpoch,
-        sstables: Vec<LocalSstableInfo>,
+        mut sstables: Vec<LocalSstableInfo>,
         sst_to_context: HashMap<HummockSstableId, HummockContextId>,
     ) -> Result<()> {
+        let mut compaction_guard = write_lock!(self, compaction).await;
         let mut versioning_guard = write_lock!(self, versioning).await;
         let _timer = start_measure_real_process_timer!(self);
         // Prevent commit new epochs if this flag is set
         if versioning_guard.disable_commit_epochs {
             return Ok(());
         }
 
-        let compaction_groups: HashMap<_, _> = self
+        let mut compaction_groups: HashMap<_, _> = self
             .compaction_group_manager
             .compaction_groups()
             .await
             .into_iter()
             .map(|group| (group.group_id(), group))
             .collect();
 
-        // Warn of table_ids that is not found in expected compaction group.
-        // It indicates:
-        // 1. Either these table_ids are never registered to any compaction group. This is FATAL
-        // since compaction filter will remove these valid states incorrectly.
-        // 2. Or the owners of these table_ids have been dropped, but their stale states are still
-        // committed. This is OK since compaction filter will remove these stale states
-        // later.
-        for (compaction_group_id, sst) in &sstables {
-            let compaction_group = compaction_groups
-                .get(compaction_group_id)
-                .ok_or(Error::InvalidCompactionGroup(*compaction_group_id))?;
-            for table_id in sst
-                .table_ids
-                .iter()
-                .filter(|t| !compaction_group.member_table_ids().contains(t))
-            {
-                tracing::warn!(
-                    "table {} doesn't belong to expected compaction group {}",
-                    table_id,
-                    compaction_group_id
-                );
-            }
-        }
+        let compaction = compaction_guard.deref_mut();
+        let versioning = versioning_guard.deref_mut();
+        let (entry_k, entry_v, mut new_hummock_version) = self
+            .sync_group(compaction, versioning, &mut compaction_groups)
+            .await?;
+        let mut new_version_delta = BTreeMapEntryTransaction::new_insert(
+            &mut versioning.hummock_version_deltas,
+            entry_k,
+            entry_v,
+        );
+        let mut branched_ssts = BTreeMapTransaction::new(&mut compaction.branched_ssts);
```

Comment:
Same here.

Reply:
> Same here.

We need to rollback, even if it's a pure in-mem structure.